### PR TITLE
Make psuedo jobs fail if dependency jobs failed in CI

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
The psuedo jobs are getting skipped when the jobs they depend on fail, and that's allowing merge with failing tests. This makes the psuedo jobs always run and check the status of the jobs they depend on.

## Testing
Testing as part of this PR by intentionally breaking one of the jobs the psuedo jobs depend on. Will revert this test break once it's verified.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ci-sdk.yaml [hunk 1]
```diff
@@ -71,6 +71,8 @@ jobs:
           run: cargo +nightly udeps
         - name: Additional per-crate checks
           run: ../tools/additional-per-crate-checks.sh ./sdk/
+        - name: Failure test
+          run: exit 1
```
Comment (crisidev) [inline .github/workflows/ci-sdk.yaml:75]: I am not sure I understand why this _have_ to fail..

**Review by crisidev — APPROVED** [review]
Nice!

File: .github/workflows/ci-sdk.yaml (hunk 1 above)
Comment (jdisanti) [reply .github/workflows/ci-sdk.yaml:75]: I'm just testing failure with that 😄 Will remove it once I know it works.

**Review by rcoh — APPROVED** [review]

File: .github/workflows/ci-sdk.yaml (hunk 1 above)
Comment (crisidev) [reply .github/workflows/ci-sdk.yaml]: Ah! Oook 😄

**Review by crisidev — APPROVED** [review]

**Pull request merged**